planner/core/task.go

```diff
@@ -2075,6 +2075,20 @@ func (p *PhysicalHashAgg) cpuCostDivisor(hasDistinct bool) (float64, float64) {
 	return math.Min(float64(finalCon), float64(partialCon)), float64(finalCon + partialCon)
 }
 
+func (p *PhysicalHashAgg) attach2TaskForMpp1Phase(mpp *mppTask) task {
+	inputRows := mpp.count()
+	// 1-phase agg: when the partition columns can be satisfied, where the plan does not need to enforce Exchange
+	// only push down the original agg
+	proj := p.convertAvgForMPP()
+	attachPlan2Task(p.self, mpp)
+	if proj != nil {
+		attachPlan2Task(proj, mpp)
+	}
```

Comment:
MppScalar2Phase inject an projection in L:2182, does here need do so?

Reply:
@hanfei1991 Fei, do you know what is the meaning of this projection? https://github.com/pingcap/tidb/blob/master/planner/core/task.go#L2164-L2172

Reply:
that projection is because tiflash dagQueryBlock can't deal with continuous aggs, so it must add a projection to separate them, actually, it is useless, because there will be an exchange between them.

one phase agg needn't it.